src/styles/getTooltipStyles.js

```diff
@@ -0,0 +1,177 @@
+ * @returns {Object}
+ */
+export default function getTooltipStyles(
+    currentSize,
+    windowWidth,
+    xOffset,
+    yOffset,
+    componentWidth,
+    componentHeight,
+    tooltipWidth,
+    tooltipHeight,
+) {
+    // Determine if the tooltip should display below the wrapped component.
+    // If a tooltip will try to render within GUTTER_WIDTH logical pixels of the top of the screen,
+    // we'll display it beneath its wrapped component rather than above it as usual.
+    const shouldShowBelow = (yOffset - tooltipHeight) < GUTTER_WIDTH;
+
+    // Determine if we need to shift the tooltip horizontally to prevent it
+    // from displaying too near to the edge of the screen.
+    const horizontalShift = computeHorizontalShift(windowWidth, xOffset, componentWidth, tooltipWidth);
+
+    const tooltipVerticalPadding = spacing.pv1;
+    const tooltipFontSize = variables.fontSizeSmall;
+
+    return {
+        animationStyle: {
+            transform: [{
+                scale: currentSize,
+            }],
+        },
+        tooltipWrapperStyle: {
+            position: 'absolute',
+            backgroundColor: themeColors.heading,
+            borderRadius: variables.componentBorderRadiusSmall,
+            ...tooltipVerticalPadding,
+            ...spacing.ph2,
+
+            // Shift the tooltip down (+) by...
+            //
+            //   component height
+            // + (pointer height - 1)
+            //
+            // OR
+            //
+            // Shift the tooltip up (-) by...
+            //   font height
+            // + top padding
+            // + bottom padding
+            // + upward shift of the pointer (pointer height)
+            // + (pointer height - 1)
+            //
+            top: shouldShowBelow
+                ? componentHeight + (POINTER_HEIGHT - 1)
+                : -1 * (
+                    tooltipFontSize
+                    + tooltipVerticalPadding.paddingTop
+                    + tooltipVerticalPadding.paddingBottom
+                    + ((2 * POINTER_HEIGHT) - 1)
+                ),
+
+            // Shift the tooltip to the left by...
+            //   half the component's width
+            // + half the tooltip's width
+            // + any horizontal shift
+            left: (tooltipWidth / -2) + (componentWidth / 2) + horizontalShift,
+        },
+        tooltipTextStyle: {
+            color: themeColors.textReversed,
+            fontFamily: fontFamily.GTA,
+            fontSize: tooltipFontSize,
+        },
+        pointerWrapperStyle: {
+            position: 'absolute',
+
+            // Shift the pointer up by its height
+            // OR
+            // Down by the component's height - its height
+            top: shouldShowBelow ? (componentHeight - POINTER_HEIGHT) : (-1 * POINTER_HEIGHT),
+
+            // Shift the pointer to the right (to the middle of the wrapped element)
+            left: (componentWidth / 2) - (POINTER_WIDTH / 2),
```

Comment:
Pretty much similar comments as before. Would be nice to add brief context for what the visual end result will be, why we are doing things, and how the math supports it etc.